rmf_fleet_adapter_python/src/battery/battery.cpp

```diff
@@ -0,0 +1,94 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/operators.h>
+#include <pybind11/iostream.h>
+#include <pybind11/eigen.h>
+#include <pybind11/chrono.h>
+#include <pybind11/stl.h>
+#include <optional>
+
+#include <memory>
+#include <random>
+#include <cstdint>
+
+#include <rmf_battery/agv/BatterySystem.hpp>
+#include <rmf_battery/agv/SimpleMotionPowerSink.hpp>
+#include <rmf_battery/agv/SimpleDevicePowerSink.hpp>
+
+namespace py = pybind11;
+namespace agv = rmf_battery::agv;
+
+void bind_battery(py::module &m)
+{
+  auto m_battery = m.def_submodule("battery");
+
+  py::class_<agv::BatterySystem>(m_battery, "BatterySystem")
+      .def_static("make", &agv::BatterySystem::make,
+                  py::arg("nominal_voltage"),
+                  py::arg("capacity"),
+                  py::arg("charging_current"))
+      .def_property_readonly("get_nominal_voltage",
+        &agv::BatterySystem::nominal_voltage)
+      .def_property_readonly("get_capacity",
+        &agv::BatterySystem::capacity)
+      .def_property_readonly("get_charging_current",
+        &agv::BatterySystem::charging_current);
+
+  // motion sink
+  py::class_<agv::SimpleMotionPowerSink>(m_battery, "SimpleMotionPowerSink")
+      // NOTE: MechanicalSystem is being abstracted
+      // .def(py::init<agv::BatterySystem&, agv::MechanicalSystem&>())
+      .def(py::init(
```

Comment:
This does not seem to reflect the function signature in `rmf_battery::agv::SimpleMotionPowerSink`. Shouldn't we also be writing bindings for `rmf_battery::agv::MechanicalSystem` and `rmf_battery::agv::PowerSystem`?

Reply:
Here, the `MechanicalSystem` and `PowerSystem` are both being abstracted within `SimpleMotionPowerSink` and `SimpleDevicePowerSink` respectively. This reduces the number of class, simplied the setup of battery for the user. e.g: 
```
battery_sys = battery.BatterySystem.make(24.0, 40.0, 8.8)
motion_sink = battery.SimpleMotionPowerSink(battery_sys, 70.0, 40.0, 0.22)
ambient_sink = battery.SimpleDevicePowerSink(battery_sys, 20.0)
tool_sink = battery.SimpleDevicePowerSink(battery_sys, 10.0)
```

Reply:
I understand the intention here. But the purpose of the bindings would be to mimic the original C++ API as closely as possible. These classes may simply be containers for physical properties, but they could be expanded in the future to have other functions. Another point is that if the documentation here is lacking, the user may refer to the C++ documentation for these classes. So we do not want to have any discrepancy between the two.

Reply:
Updated: https://github.com/osrf/rmf_fleet_adapter_python/pull/10/commits/e6db6ca49713634c0c93e5fa12987faefcb31841